### PR TITLE
GEODE-9486: Improve AnalyzeSerializables integration tests

### PR DESCRIPTION
**Pull request by kirklund**
Allow geode modules to have an AnalyzeSerializables integration
test without implementing SanctionedSerializablesService. 

Improve debugging information for AnalyzeSerializables integration
tests:
- Provide new failure message when no SanctionedSerializablesService
exists.
- Create ANALYZE_SERIALIZABLES.md to provide detailed instructions for
failures involving AnalyzeSerializables integration tests.
- Reference ANALYZE_SERIALIZABLES.md in failure messages.

Remove geode-serialization and geode-common jars from geode-pulse
.war file:
- Change getModuleClass() to return Optional.
- Delete PulseSanctionedSerializablesService and its resources.
- Change geode-pulse dependency on geode-serialization to be for
integrationTest only.

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced in the commit message?

- [ ] Has your PR been rebased against the latest commit within the target branch (typically `develop`)?

- [ ] Is your initial contribution a single, squashed commit?

- [ ] Does `gradlew build` run cleanly?

- [ ] Have you written or updated unit tests to verify your changes?

- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?

### CODE REVIEW

**Review by onichols-pivotal — APPROVED** [review]

File: geode-serialization/ANALYZE_SERIALIZABLES.md [hunk 1]
```diff
@@ -0,0 +1,129 @@
...
+The content of the failure message depends on whether the module implements 
+SanctionedSerializablesService.
+
+If it implements SanctionedSerializablesService, the failure message looks like:
+```
+New or moved classes----------------------------------------
+org/apache/geode/CancelException,true,3215578659523282642
+
+If the class is not persisted or sent over the wire, add it to the file
+    /Users/klund/dev/geode3/geode-core/src/integrationTest/resources/org/apache/geode/codeAnalysis/excludedClasses.txt
```
Comment (demery-pivotal) [inline geode-serialization/ANALYZE_SERIALIZABLES.md:67]: Do you want your username immortalized in these file paths?

File: geode-serialization/ANALYZE_SERIALIZABLES.md [hunk 2]
```diff
@@ -0,0 +1,129 @@
...
+Otherwise, if this doesn't break backward compatibility, copy the file
+    /Users/klund/dev/geode3/geode-core/build/integrationTest/test-worker-000009/actualSerializables.dat
+    to 
+    /Users/klund/dev/geode3/geode-core/src/main/resources/org/apache/geode/internal/sanctioned-geode-core-serializables.txt
+If this potentially breaks backward compatibility, follow the instructions in
+    README.MD
```
Comment (demery-pivotal) [inline geode-serialization/ANALYZE_SERIALIZABLES.md]: Change `README.MD` to the new path to the instructions.

**Review by demery-pivotal — APPROVED** [review]

File: geode-serialization/ANALYZE_SERIALIZABLES.md (hunk 1 above)
Comment (kirklund) [reply geode-serialization/ANALYZE_SERIALIZABLES.md:67]: No, I'll remove it

**Review by mhansonp — APPROVED** [review]

**Review by boglesby — APPROVED** [review]

**Review by jinmeiliao — APPROVED** [review]

**Comment by jchen21** [comment]
I recommend putting a link to `ANALYZE_SERIALIZABLES.md` in one of the `md` files in the Geode root directory, where you see appropriate. So that it is easy to find the new instructions you added. Otherwise, people might not be aware of the existence of this helpful doc.

**Review by jchen21 — APPROVED** [review]

**Review by agingade — APPROVED** [review]

**Review by dschneider-pivotal — APPROVED** [review]

**Review by echobravopapa — APPROVED** [review]

File: geode-serialization/ANALYZE_SERIALIZABLES.md [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+The content of the failure message depends on whether the module implements `SanctionedSerializablesService`.
+
+If it implements `SanctionedSerializablesService`, the failure message looks like:
+```
+New or moved classes----------------------------------------
+org/apache/geode/CancelException,true,3215578659523282642
+
+If the class is not persisted or sent over the wire, add it to the file
+    /path/to/geode/geode-core/src/integrationTest/resources/org/apache/geode/codeAnalysis/excludedClasses.txt
+Otherwise, if this doesn't break backward compatibility, copy the file
```
Comment (gesterzhou) [inline geode-serialization/ANALYZE_SERIALIZABLES.md:61]: "doesn't"?

I wonder if it's typo. Should it be "does" here?

**Review by gesterzhou — APPROVED** [review]

**Review by nabarunnag — APPROVED** [review]

**Review by jdeppe-pivotal — APPROVED** [review]

**Pull request merged**